### PR TITLE
Migrate to Pressable.

### LINKED ISSUE #2939: [RFC]: Migration of all press interactions to Pressable Component
**Description**
Pressable is a Core Component wrapper that can detect various stages of press interactions on any of its defined children.

**Why?**
As per the official React Native docs -> https://reactnative.dev/docs/touchablehighlight 
> If you're looking for a more extensive and future-proof way to handle touch-based input, check out the Pressable API.

This is a generic component and is the one-stop solution for all the press interactions.

Let's Discuss 📢

### PR DESCRIPTION
**Pull request by arpitBhalla**
**What kind of change does this PR introduce?**
Convert Touchable(Highlight|Opacity) based components to Pressable.  

- [x] Avatar
- [x] Accessary
- [x] Badge
- [x] ButtonGroup
- [x] SpeedDial
- [x] Checkbox
- [x] Icon
- [x] Image
- [x] SearchBar
- [x] Social Icon
- [x] Overlay
- [x] FeaturedTile
- [x] Tile
- [x] ListItem

**Did you add tests for your changes?**
Yes, I have updated the tests and migrated them to RN-testing Library.

**If relevant, did you update the documentation?**
No changes are required.

**Does this PR introduce a breaking change?**
No, this would not break anything.

**Other information**
GSoC Task of @arpitBhalla.
- [x] Self Review for any kind of error in code.

#### I have tested the PR manually in demo app / Snack
- [x] Android
- [ ] iOS
- [x] Web 

Closes #2939

### CODE REVIEW

File: src/ButtonGroup/ButtonGroup.tsx [hunk 1]
```diff
@@ -176,7 +178,14 @@ export const ButtonGroup: RneFunctionComponent<ButtonGroupProps> = ({
...
               onShowUnderlay={onShowUnderlay}
-              underlayColor={underlayColor}
+              android_ripple={{
+                color: Color(underlayColor)
+                  .alpha(activeOpacity)
+                  .rgb()
+                  .toString(),
+                borderless: false,
+                radius: -5,
+              }}
```
Comment (pranshuchittora) [inline src/ButtonGroup/ButtonGroup.tsx:188]: Let's extract this out as a seperate util

File: src/CheckBox/CheckBox.tsx [hunk 2]
```diff
@@ -70,7 +70,7 @@ export const CheckBox: RneFunctionComponent<CheckBoxProps> = ({
   onPress,
   onLongPress,
   checkedColor = theme?.colors?.primary,
-  ...rest
+  ...props
```
Comment (pranshuchittora) [inline src/CheckBox/CheckBox.tsx:73]: Should be rest only

**Comment by pranshuchittora** [comment]
Reviewed today. LGTM. 
Will merge this after getting approval on #3150

File: src/Badge/Badge.tsx [hunk 3]
```diff
@@ -44,11 +48,12 @@ export const Badge: RneFunctionComponent<BadgeProps> = ({
...
   badgeStyle,
   onPress,
-  Component = onPress ? TouchableOpacity : View,
+  Component = onPress ? Pressable : View,
   value,
   theme,
   status = 'primary',
-  ...attributes
+  pressableProps,
+  ...props
```
Comment (pranshuchittora) [inline src/Badge/Badge.tsx:56]:
```suggestion
  ...rest
```

Pls make this change at all places

File: src/Chip/__tests__/__snapshots__/Chip.test.tsx.snap [hunk 4]
```diff
@@ -1,83 +1,18 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
-exports[`Chip Component should render outline 1`] = `
+exports[`Chip Component should render without issues 1`] = `
```
Comment (pranshuchittora) [inline src/Chip/__tests__/__snapshots__/Chip.test.tsx.snap:3]: remove without issues

File: src/Tile/__tests__/__snapshots__/Tile.tsx.snap [hunk 5]
```diff
@@ -404,12 +421,18 @@ exports[`Tile component should render tile with icon 1`] = `
...
-<ForwardRef
-  activeOpacity={0.9}
+<Pressable
+  android_ripple={
+    Object {
+      "borderless": false,
+      "color": "rgba(0, 0, 0, 0.9)",
+      "radius": -5,
+    }
+  }
```
Comment (pranshuchittora) [inline src/Tile/__tests__/__snapshots__/Tile.tsx.snap:435]: Provide a way to override the `android_ripple`

File: src/Avatar/Avatar.tsx [hunk 6]
```diff
@@ -81,14 +85,18 @@ export type AvatarBaseProps = {
 
   /** Custom ImageComponent for Avatar. */
   ImageComponent?: React.ComponentClass;
-};
+} & InlinePressableProps;
 
 /** Avatars are found all over ui design from lists to profile screens.
  * They are commonly used to represent a user and can contain photos, icons, or even text. */
 export const AvatarBase: RneFunctionComponent<AvatarBaseProps> = ({
```
Comment (pranshuchittora) [inline src/Avatar/Avatar.tsx:92]: You forgot to add `onPressIn` and `onPressOut` to `AvatarBaseProps`

File: src/Badge/Badge.tsx [hunk 7]
```diff
@@ -27,15 +31,12 @@ export type BadgeProps = {
   /** Text value to be displayed by badge, defaults to empty. */
   value?: React.ReactNode;
 
-  /** Function called when pressed on the badge. */
-  onPress?: (...args: any[]) => any;
-
```
Comment (pranshuchittora) [inline src/Badge/Badge.tsx:32]: Pls updated the props over here as well.

**Review by pranshuchittora — CHANGES_REQUESTED** [review]
Pls make sure that you have updated all the props.

File: src/Avatar/Avatar.tsx (hunk 6 above)
Comment (arpitBhalla) [reply src/Avatar/Avatar.tsx:92]: `InlinePressableProps` this includes onPressIn and onPressOut

File: src/Badge/Badge.tsx (hunk 7 above)
Comment (arpitBhalla) [reply src/Badge/Badge.tsx:32]: `InlinePressableProps` this includes onPressIn and onPressOut

**Comment by flyingcircle** [comment]
Will need to resolve conflicts here.

**Comment by arpitBhalla** [comment]
@flyingcircle Can you first review all Test PRs

**Comment by pranshuchittora** [comment]
@arpit-bhalla does this PR have any dependency?

**Comment by arpitBhalla** [comment]
@khushal87 Tests are failing but Tests CI is passing, can you look into this please.

**Comment by khushal87** [comment]
> @khushal87 Tests are failing but Tests CI is passing, can you look into this please.
> 
> ```
> [error]: update and commit docs Error: Command failed: yarn docs:build
> 
> /home/arpit/Desktop/os/rne/website/scripts/docgen/docgenParser.ts:137
>       prop.defaultValue.value = platformMappingHandler(prop.defaultValue.value);
>                                                                          ^
> TypeError: Cannot read property 'value' of null
>     at propFilter (/home/arpit/Desktop/os/rne/website/scripts/docgen/docgenParser.ts:137:74)
>     at Parser.propFilter (/home/arpit/Desktop/os/rne/node_modules/react-docgen-typescript/src/buildFilter.ts:21:20)
>     at Parser.getComponentInfo (/home/arpit/Desktop/os/rne/node_modules/react-docgen-typescript/src/parser.ts:372:19)
>     at /home/arpit/Desktop/os/rne/node_modules/react-docgen-typescript/src/parser.ts:1379:28
>     at Array.forEach (<anonymous>)
>     at /home/arpit/Desktop/os/rne/node_modules/react-docgen-typescript/src/parser.ts:1378:18
>     at Array.reduce (<anonymous>)
>     at parseWithProgramProvider (/home/arpit/Desktop/os/rne/node_modules/react-docgen-typescript/src/parser.ts:1367:6)
>     at Object.parse (/home/arpit/Desktop/os/rne/node_modules/react-docgen-typescript/src/parser.ts:183:14)
>     at Object.generateComponentDocs (/home/arpit/Desktop/os/rne/website/scripts/docgen/generateComponentDocs.ts:8:6)
> error Command failed with exit code 1.
> ```

As you have changed the type of `ButtonGroup`'s type `Component`. So we don't need this condition anymore in `website/scripts/docgenParser.ts`:
```tsx
    // To deal with the platform specific props default value in ButtonGroup
    // Input - Platform.select<typeof React.Component>({
    //   android: TouchableNativeFeedback,
    //   default: TouchableOpacity,
    // })
    // Output - TouchableOpacity(ios),TouchableNativeFeedback(android),TouchableOpacity(web)
    if (component.name === 'ButtonGroup' && prop.name === 'Component') {
      prop.defaultValue.value = platformMappingHandler(prop.defaultValue.value);
    }
```

File: src/SocialIcon/SocialIcon.tsx [hunk 8]
```diff
@@ -165,24 +163,37 @@ export const SocialIcon: RneFunctionComponent<SocialIconProps> = ({
   loading,
   onLongPress,
   onPress,
-  Component = onPress || onLongPress ? TouchableHighlight : View,
+  onPressOut,
+  onPressIn,
+  Component = onPress || onLongPress || onPressIn || onPressOut
```
Comment (khushal87) [inline src/SocialIcon/SocialIcon.tsx:168]: Please deal with this case in `website/scripts/docgenParser.ts` @arpitBhalla

File: src/Image/Image.tsx [hunk 9]
```diff
@@ -7,51 +7,56 @@ import {
...
 images with a placeholder and smooth image load transitioning. */
 export const Image: RneFunctionComponent<ImageProps> = ({
   onPress,
   onLongPress,
-  Component = onPress || onLongPress ? TouchableOpacity : View,
+  onPressIn,
+  onPressOut,
+  Component = onPress || onLongPress || onPressIn || onPressOut
+    ? Pressable
+    : View,
```
Comment (khushal87) [inline src/Image/Image.tsx:59]: Also with this.

**Comment by khushal87** [comment]
Also, deal with types of setOpacityTo, onPress in ButtonGroup

File: src/Badge/Badge.tsx [hunk 10]
```diff
@@ -44,11 +45,17 @@ export const Badge: RneFunctionComponent<BadgeProps> = ({
   textProps,
   badgeStyle,
   onPress,
-  Component = onPress ? TouchableOpacity : View,
+  onLongPress,
+  onPressOut,
+  onPressIn,
+  Component = onPress || onLongPress || onPressIn || onPressOut
+    ? Pressable
+    : View,
```
Comment (khushal87) [inline src/Badge/Badge.tsx:53]: Deal with this too.

**Comment by khushal87** [comment]
onLongPress in Avatar.md

File: src/SocialIcon/SocialIcon.tsx (hunk 8 above)
Comment (arpitBhalla) [reply src/SocialIcon/SocialIcon.tsx:168]: How?

File: src/Image/Image.tsx (hunk 9 above)
Comment (arpitBhalla) [reply src/Image/Image.tsx:59]: How?

File: src/Badge/Badge.tsx (hunk 10 above)
Comment (arpitBhalla) [reply src/Badge/Badge.tsx:53]: How?

Comment (arpitBhalla) [reply src/Badge/Badge.tsx:53]: How?

File: src/SocialIcon/SocialIcon.tsx (hunk 8 above)
Comment (khushal87) [reply src/SocialIcon/SocialIcon.tsx:168]: Add conditions, as I have added for other props.

**Pull request merged**